cli/start.go

```diff
@@ -264,11 +265,18 @@ func runStart(_ *cobra.Command, _ []string) error {
 
 // rerunBackground restarts the current process in the background.
 func rerunBackground() error {
-	args := append([]string(nil), os.Args...)
-	// TODO(bdarnell): this looks silly in ps. Try to remove the
-	// `--background` flag instead of adding a second flag to reverse
-	// it.
-	args = append(args, "--background=false")
+	args := make([]string, 0, len(os.Args))
+	foundBackground := false
+	for _, arg := range os.Args {
+		if arg == "--background" || strings.HasPrefix(arg, "--background=") {
+			foundBackground = true
+			continue
+		}
+		args = append(args, arg)
+	}
+	if !foundBackground {
+		args = append(args, "--background=false")
```

Comment:
If we're in `rerunBackground`, then the background flag must have been present in a way that the flag package was able to parse, but our simplistic prefix matching didn't find it. This is a safeguard against the flag parsing rules being more complicated than we understand.

One thing we're missing here is that the string `--background` could be used as the argument to another flag: `cockroach start --store --background --background` is legal (if nonsensical) and creates a data directory named `--background` while running in the background. This code would strip out both occurrences of `--background`.

This sort of parsing bug sets of my security senses, although I'm not able to come up with any problematic scenarios other than someone inflicting pain on themselves by deliberately choosing weird names for things.